cli/lib.rs

```diff
@@ -297,6 +297,27 @@ fn eval_command(flags: DenoFlags, argv: Vec<String>) {
   tokio_util::run(main_future);
 }
 
+fn doc_command(flags: DenoFlags, argv: Vec<String>) {
+  let (mut _worker, state) = create_worker_and_state(flags, argv);
+
+  let main_module = state.main_module().unwrap();
+  assert!(state.argv.len() >= 2);
+  debug!(">>>>> doc_async START");
+  let doc_future = state
+    .ts_compiler
+    .doc_async(state.clone(), main_module.to_string())
+    .map_err(|err| {
+      debug!("diagnostics returned, exiting!");
+      eprintln!("");
+      print_err_and_exit(err);
+    })
+    .and_then(move |_| {
+      debug!("<<<<< bundle_async END");
+      Ok(())
+    });
+  tokio_util::run(doc_future);
+}
+
```

Comment:
Courtesy note: with addition of eager poll this will need to be called alike to `bundle_command`, ie. polling worker and only then doing `doc_future`:
https://github.com/denoland/deno/blob/4221b90c3febbe03a4b47e47248263741a0fdd4a/cli/lib.rs#L307-L325